ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/TagWithLedgerEndImpl.scala

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.sandbox.bridge
+
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.IndexService
+import com.daml.ledger.sandbox.bridge.ConflictCheckingLedgerBridge._
+import com.daml.metrics.Timed
+import com.daml.platform.ApiOffset
+
+import scala.concurrent.{ExecutionContext, Future}
+
+private[bridge] class TagWithLedgerEndImpl(
+    indexService: IndexService,
+    bridgeMetrics: BridgeMetrics,
+)(implicit executionContext: ExecutionContext)
+    extends TagWithLedgerEnd {
+  override def apply(
+      preparedSubmission: Validation[PreparedSubmission]
+  ): AsyncValidation[(Offset, PreparedSubmission)] = preparedSubmission match {
+    case Left(rejection) => Future.successful(Left(rejection))
+    case Right(preparedSubmission) =>
+      Timed.future(
+        bridgeMetrics.Stages.tagWithLedgerEnd,
```

Comment:
Doesn't this always complete instantly? `LedgerBackedIndexService.currentLedgerEnd` unconditionally returns a Future.successful.

Reply:
You are right. It is a relic from the benchmarking period. It was the threadpool congestion probe: it used to have up to 1ms duration on loaded participant -> high congestion. I'll add a TODO to remove.